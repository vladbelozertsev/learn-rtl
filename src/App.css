body {
  padding: 25px;
  background-color: #3366ff;
  color: #fff;
  font-size: 20px;
}

input {
  padding: 10px;
  height: 30px;
  border: none;
  outline: none;
  border-radius: 5px;
  margin-left: 15px;
}
