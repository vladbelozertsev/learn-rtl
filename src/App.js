import { useState, useEffect } from 'react';
import './App.css';

const Search = (props) => {
  return (
    <div>
      <label htmlFor="search">{props.children}</label>
      <input type="text" id="search" defaultValue={props.value} onChange={props.onChange} />
    </div>
  );
};

const App = () => {
  const [search, setSearch] = useState('');

  const handleChange = (e) => {
    setSearch(e.target.value);
  };

  useEffect(() => {
    console.log(search);
  }, [search]);

  return (
    <div>
      <Search value={search} onChange={handleChange}>
        Search:
      </Search>
      <p>Searches for {search ? search : '...'}</p>
    </div>
  );
};

export default App;
