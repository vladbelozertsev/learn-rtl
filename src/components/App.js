import '../App.css';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import SearchTypes from './SearchTypes';
import SearchVariants from './SearchVariants';
import Nav from './Nav';

export default function App() {
  return (
    <Router>
      <div>
        <h2>Перейти к уроку</h2>
        <Nav />
        <Switch>
          <Route path="/search-types">
            <SearchTypes />
          </Route>
          <Route path="/search-variants">
            <SearchVariants />
          </Route>
        </Switch>
      </div>
    </Router>
  );
}
