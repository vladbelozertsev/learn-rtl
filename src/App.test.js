import { render, screen } from '@testing-library/react';
import App from './App';

describe('App', () => {
  // Поиск по тексту в разметке:
  it('render App component and search by text', () => {
    render(<App />);
    screen.debug(); // - в случае ошибки выводит в консоль HTML-разметку соответствующего компонента (см render(...))
    expect(screen.getByText(/Search:/i)).toBeInTheDocument();
    expect(screen.getByText(/Searches for/i)).toBeInTheDocument();
  });

  // Поиск по ролям:
  // https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles
  it('render App component and search by role', () => {
    render(<App />);
    screen.debug(); // - в случае ошибки выводит в консоль HTML-разметку соответствующего компонента (см render(...))
    expect(screen.getByRole('textbox')).toBeInTheDocument();
    expect(screen.getByRole('button')).toBeInTheDocument();
    // Роль формы должна быть определена явно, через aria-label атрибут:
    // https://github.com/testing-library/dom-testing-library/issues/474
    expect(screen.getByRole('form')).toBeInTheDocument();
  });

  it('render App component and search by other methods', () => {
    render(<App />);
    screen.debug(); // - в случае ошибки выводит в консоль HTML-разметку соответствующего компонента (см render(...))
    expect(screen.getByPlaceholderText('search text...')).toBeInTheDocument();
    expect(screen.getByAltText('search image')).toBeInTheDocument();
    expect(screen.getByDisplayValue('yoyo')).toBeInTheDocument();
  });
});
