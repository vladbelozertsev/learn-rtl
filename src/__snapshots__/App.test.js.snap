// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div>
    <div>
      <label
        for="search"
      >
        Search:
      </label>
      <input
        id="search"
        type="text"
        value=""
      />
    </div>
    <p>
      Searches for ...
    </p>
  </div>
</DocumentFragment>
`;
